File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/tools/JSON.java
package com.valleapp.valletpvlib.tools;

import android.content.Context;
import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class JSON {

    public void serializar(String file, JSONObject obj, Context context){
        try{
            OutputStreamWriter fos = new OutputStreamWriter(context.openFileOutput(file, Context.MODE_PRIVATE));
            fos.write(obj.toString());
            fos.close();
        }catch(Exception e){
            Log.e("JSON", e.toString());
        }
    }

    public JSONObject deserializar(String file, Context context) throws JSONException {
        StringBuilder strJSON = new StringBuilder();

        try{
            BufferedReader fos = new BufferedReader(new InputStreamReader(context.openFileInput(file)));
            String tmp = fos.readLine();
            while (tmp!=null) {
                strJSON.append(tmp);
                tmp = fos.readLine();
            }
            fos.close();
        }catch(Exception e){
            Log.e("JSON", e.toString());
        }

        if (strJSON.length() == 0) return null;
        else {
            return new JSONObject(strJSON.toString());
        }

    }
}

File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/tools/JSON.kt
package com.valleapp.valletpvlib.tools

import android.content.Context
import android.util.Log
import org.json.JSONException
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.OutputStreamWriter

class JSON {

    fun serializar(file: String, obj: JSONObject?, context: Context?) {
        try {
            val fos = OutputStreamWriter(context?.openFileOutput(file, Context.MODE_PRIVATE))
            fos.write(obj.toString())
            fos.close()
        } catch (e: Exception) {
            Log.e("JSON", e.toString())
        }
    }

    @Throws(JSONException::class)
    fun deserializar(file: String, context: Context?): JSONObject? {
        val strJSON = StringBuilder()

        try {
            val fos = BufferedReader(InputStreamReader(context?.openFileInput(file)))
            var tmp = fos.readLine()
            while (tmp != null) {
                strJSON.append(tmp)
                tmp = fos.readLine()
            }
            fos.close()
        } catch (e: Exception) {
            Log.e("JSON", e.toString())
        }

        return if (strJSON.isEmpty()) null else JSONObject(strJSON.toString())
    }
}
